docker/Dockerfile.nvidia

```diff
@@ -129,59 +88,17 @@ RUN if [ "x$MPI_VER" = "x4" ]; then \
                $HPCSDK_HOME/comm_libs/mpi ; \
     fi;
 
-# Install pip dependencies and devito as a pip package
-RUN python3 -m venv /venv && \
-    /venv/bin/pip install --no-cache-dir --upgrade pip && \
-    /venv/bin/pip install --no-cache-dir wheel && \
-    /venv/bin/pip install --no-cache-dir -e /app/devito[extras,nvidia] && \
-    # Need extra CFLAGS for mpi4py
-    CFLAGS=-noswitcherror /venv/bin/pip install --no-cache-dir -r /app/devito/requirements-mpi.txt && \
-    rm -rf ~/.cache/pip
-
-ADD docker/run-jupyterlab.sh /jupyter
-ADD docker/run-tests.sh /tests
-ADD docker/run-print-defaults.sh /print-defaults
-ADD docker/entrypoint.sh /docker-entrypoint.sh
-ADD docker/nvdashboard.json /app/nvdashboard.json
+# Make devito env vars file
+RUN echo '#!/usr/bin/env bash' > /app/setup-devito-env.sh && \
+    echo 'export DEVITO_ARCH="nvc"' >> /app/setup-devito-env.sh && \
+    echo 'export DEVITO_PLATFORM="nvidiaX"' >> /app/setup-devito-env.sh && \
+    echo 'export DEVITO_LANGUAGE="openacc"' >> /app/setup-devito-env.sh
 
-RUN chmod +x /print-defaults /jupyter /tests /docker-entrypoint.sh && \
-    /venv/bin/jupyter serverextension enable dask_labextension 
-#    /venv/bin/jupyter lab workspaces import /app/nvdashboard.json
+# Install python nvidia dependencies
+RUN /venv/bin/pip install --no-cache-dir --no-cache-dir --upgrade pip && \
+    /venv/bin/pip install --no-cache-dir --no-cache-dir -r https://raw.githubusercontent.com/devitocodes/devito/master/requirements-nvidia.txt
 
-## Environment Variables for OpenACC Builds
-# Reference: https://github.com/devitocodes/devito/wiki/FAQ#can-i-manually-modify-the-c-code-generated-by-devito-and-test-these-modifications
-# Set arch to PGI (pgcc)
-ENV DEVITO_ARCH="nvc" 
-ENV DEVITO_LANGUAGE="openacc"
-ENV DEVITO_PLATFORM=nvidiaX
-# Options: [unset, 1] For PGI openacc; Should only be set after a first execution of the benchmark
-# ENV DEVITO_JIT_BACKDOOR=1 
-
-# Enable logging, Options: [unset, PERF, DEBUG]
-ENV DEVITO_LOGGING=DEBUG 
-#ENV PGI_ACC_TIME=1 
-
-## Create App user 
-# Set the home directory to our app user's home.
-ENV HOME=/app
-ENV APP_HOME=/app
-
-# Create the home directory for the new app user.
-# Create an app user so our program doesn't run as root.
-# Chown all the files to the app user.
-RUN mkdir -p /app && \
-    groupadd -r app && \
-    useradd -r -g app -d /app -s /sbin/nologin -c "Docker image user" app && \
-    chown -R app:app $APP_HOME && \
-    chown app:app /docker-entrypoint.sh && \
-    chown app:app /print-defaults && \
-    chown app:app /tests && \
-    chown app:app /jupyter
-    
-# Change to the app user.
-USER app
-WORKDIR /app/devito
+ENV CFLAGS=-noswitcherror 
 
 EXPOSE 8888
-ENTRYPOINT ["/docker-entrypoint.sh"]
-CMD ["/jupyter"]
+CMD ["/bin/bash"]
```

Comment:
Better yes IMO

Reply:
That's because it's the compiler only image, the devito image still has the entrypoint